great_expectations/core/configuration.py

```diff
@@ -1,18 +1,29 @@
 """Contains general abstract or base classes used across configuration objects."""
+import copy
 from abc import ABC
 from typing import Optional
 
+from great_expectations.marshmallow__shade.decorators import post_dump
+from great_expectations.marshmallow__shade.schema import Schema
 from great_expectations.types import SerializableDictDot
 
 
 class AbstractConfig(ABC, SerializableDictDot):
     """Abstract base class for Config objects. Sets the fields that must be included on a Config."""
 
     def __init__(self, id_: Optional[str] = None, name: Optional[str] = None) -> None:
-        # Note: name and id are optional currently to avoid updating all documentation within
-        # the scope of this work.
-        if id_ is not None:
-            self.id_ = id_
-        if name is not None:
-            self.name = name
+        self.id_ = id_
+        self.name = name
         super().__init__()
+
+
+class AbstractConfigSchema(Schema):
+    REMOVE_KEYS_IF_NONE = ["id", "name"]
+
+    @post_dump
+    def filter_none(self, data: dict, **kwargs) -> dict:
+        data = copy.deepcopy(data)
```

Comment:
we don't need a deepcopy here; consider replacing with a comprehension ex:

return {
            key: value for key, value in data.items()
            ...
        }

Reply:
done!